cli/src/semgrep/rule_match.py

```diff
@@ -221,8 +221,16 @@ def get_cli_unique_key(self) -> CliUniqueKey:
             # current scan results if an issue arises.
             self.annotated_rule_name if self.from_transient_scan else self.rule_id,
             str(self.path),
-            self.start.offset,
-            self.end.offset,
+            # Lockfile-only supply chain findings don't have offsets (not for any technical reason, the parsers just aren't set up to output them, though of course they could be)
+            # So just use line numbers instead
+            # TODO: remove this when the switch to ocaml is done
+            *(
+                (self.start.offset, self.end.offset)
+                if not (
+                    self.extra.get("sca_info") and not self.extra["sca_info"].reachable
+                )
+                else (self.start.line, self.end.line)
+            ),
```

Comment:
The reliance on self.extra.get("sca_info") here scares me a little bit, because it feels like an implementation detail that should be abstracted. Can you use self.product == SCA here in case how we implement that later changes?

Also, the double negative here is messing with my head. Is this the same thing and maybe easier to read?

```suggestion
                (self.start.line, self.end.line)
                if (
                    self.extra.get("sca_info") and not self.extra["sca_info"].reachable
                )
                else (self.start.offset, self.end.offset)
            ),
```

Reply:
I did it in that order because I wanted the "default" case to be textually first, but I agree the logic is a bit confusing, I'll go with your suggestion